### PR TITLE
cleanup/ipam: Remove hostscope-legacy IPAM option

### LINKED ISSUE #12974: Clean up hostscope-legacy ipam option
Related to https://github.com/cilium/cilium/blob/master/pkg/ipam/option/option.go#L18-L20

### PR DESCRIPTION
**Pull request by sayboras**
To remove hostscope-legacy IPAM option

Fixes: #12974

Signed-off-by: Tam Mach <sayboras@yahoo.com>

```release-note
cleanup/ipam: Remove hostscope-legacy IPAM option
```

TODO:
- [x] Update docs
- [ ] Test upgrade path

### CODE REVIEW

**Comment by sayboras** [comment]
test-me-please

**Comment by sayboras** [comment]
test-me-please

**Comment by sayboras** [comment]
test-me-please

File: Documentation/install/upgrade.rst [hunk 1]
```diff
@@ -369,6 +369,9 @@ Removed options
 * The ``prometheus-serve-addr-deprecated`` option is now removed. Please use
   ``prometheus-serve-addr`` instead.
 
+* The ``hostscope-legacy`` option value for ``ipam`` is not removed. Defaults to
```
Comment (joestringer) [inline Documentation/install/upgrade.rst]:
```suggestion
* The ``hostscope-legacy`` option value for ``ipam`` is now removed. Defaults to
```

File: Documentation/install/upgrade.rst [hunk 2]
```diff
@@ -393,6 +396,9 @@ Removed cilium-operator options
 * The ``metrics-address`` option is now removed. Please use
   ``operator-prometheus-serve-addr`` instead.
 
+* The ``hostscope-legacy`` option value for ``ipam`` is not removed.
```
Comment (joestringer) [inline Documentation/install/upgrade.rst]:
```suggestion
* The ``hostscope-legacy`` option value for ``ipam`` is now removed.
```

File: Documentation/install/upgrade.rst (hunk 1 above)
Comment (joestringer) [reply Documentation/install/upgrade.rst]: Also maybe to be a bit more clear?
````
The ``ipam`` option now defaults to ``cluster-pool``
````

**Review by joestringer — CHANGES_REQUESTED** [review]
Minor docs typos/nits, otherwise LGTM

**Comment by sayboras** [comment]
Jenkins jobs are completed successfully, force push to address review comments related to docs.

**Review by aanm — APPROVED** [review]

**Comment by aanm** [comment]
test-me-please

**Comment by sayboras** [comment]
Failure seems related to https://github.com/cilium/cilium/issues/13056

```
20:17:43  Summarizing 1 Failure:
20:17:43  
20:17:43  [Fail] K8sBandwidthTest Checks Bandwidth Rate-Limiting [It] Checks Pod to Pod bandwidth, geneve tunneling 
20:17:43  /home/jenkins/workspace/Cilium-PR-K8s-1.12-net-next/src/github.com/cilium/cilium/test/k8sT/Bandwidth.go:142
```

**Review by fristonio — CHANGES_REQUESTED** [review]
The changes in the PR looks good to me. But I wonder with the changes in the PR we don't need any documentation related to `hostscope` IPAM as well?
Like removing - https://github.com/cilium/cilium/blob/master/Documentation/concepts/networking/ipam/hostscope.rst

**Review by christarazi — APPROVED** [review]
LGTM assuming everyone else's comments will be addressed.

**Comment by sayboras** [comment]
> The changes in the PR looks good to me. But I wonder with the changes in the PR we don't need any documentation related to `hostscope` IPAM as well?
> Like removing - https://github.com/cilium/cilium/blob/master/Documentation/concepts/networking/ipam/hostscope.rst

Good catch, thanks a lot. Let me scan the docs one more time :monocle_face:

**Comment by sayboras** [comment]
Force pull for docs update, full CI is not required

<details>
<summary>git diff</summary>

```
        deleted:    concepts/networking/ipam/hostscope.rst
        modified:   concepts/networking/ipam/index.rst

diff --git a/Documentation/concepts/networking/ipam/index.rst b/Documentation/concepts/networking/ipam/index.rst
index 32d770231..4bb72e55a 100644
--- a/Documentation/concepts/networking/ipam/index.rst
+++ b/Documentation/concepts/networking/ipam/index.rst
@@ -24,5 +24,4 @@ Cilium. Various IPAM modes are supported to meet the needs of different users:
    eni
    gke
    crd
-   hostscope
    deep_dive
```

</details>

**Review by fristonio — APPROVED** [review]
LGTM :100:

**Comment by sayboras** [comment]
test-me-please

**Pull request merged**